lib/ui/signin/sign_in_form.dart

```diff
@@ -71,10 +80,14 @@ class _SignInFormState extends State<SignInForm> {
 
   _validateInputFields() {
     setState(() {
-      bool isNotNull = _emailInput != null && _passwordInput != null;
-      bool isNotEmpty =
-          _emailInput?.trim() != "" && _passwordInput?.trim() != "";
-      _isSignInButtonEnabled = isNotNull && isNotEmpty;
+      bool isNotEmpty = _emailInput.trim() != "" && _passwordInput.trim() != "";
+      bool isValidEmail = _validateEmail(_emailInput);
+      _isSignInButtonEnabled = isNotEmpty && isValidEmail;
     });
   }
+
+  bool _validateEmail(String email) {
+    RegExp emailRegex = RegExp(r'^[\w-.]+@([\w-]+\.)+[\w-]{2,4}$');
+    return emailRegex.hasMatch(email);
+  }
```

Comment:
it would be better to leave it in a Validator class so you can write test for it or add more form validation.
Alternatively, you may want to use a lib like https://github.com/TheMisir/form-validator